packages/medusa/src/subscribers/configurable-notifications.ts

```diff
@@ -0,0 +1,64 @@
+import { IEventBusService, INotificationModuleService } from "@medusajs/types"
+import { get } from "lodash"
+
+type InjectedDependencies = {
+  notificationModuleService: INotificationModuleService
+  eventBusModuleService: IEventBusService
+}
+
+// TODO: The config should be loaded dynamically from medusa-config.js
+// TODO: We can use a more powerful templating syntax to allow for eg. combining fields.
+const config = [
+  {
+    event: "order.created",
+    template: "order-created-template",
+    channel: "email",
+    to: "order.email",
+    resource_id: "order.id",
+    data: {
+      order_id: "order.id",
+    },
+  },
+]
+
+class ConfigurableNotificationsSubscriber {
+  private readonly eventBusModuleService_: IEventBusService
+  private readonly notificationModuleService_: INotificationModuleService
+
+  constructor({
+    eventBusModuleService,
+    notificationModuleService,
+  }: InjectedDependencies) {
+    this.eventBusModuleService_ = eventBusModuleService
+    this.notificationModuleService_ = notificationModuleService
+
+    config.forEach((eventHandler) => {
+      this.eventBusModuleService_.subscribe(
+        eventHandler.event,
+        async (data: any) => {
+          const payload = data.data
+
+          const notificationData = {
+            template: eventHandler.template,
+            channel: eventHandler.channel,
+            to: get(payload, eventHandler.to),
+            trigger_type: eventHandler.event,
+            resource_id: get(payload, eventHandler.resource_id),
+            data: Object.entries(eventHandler.data).reduce(
```

Comment:
todo: I would remove the use of lodash in that file

Reply:
@adrien2p a question on this, why don't we use lodash more?? 😄

Reply:
because it is yet another dependency that we dont fully use and we only need a few utils that we can carry ourselves

Reply:
or just install the get 😄

Reply:
I think we can use more of it if it was installed everywhere, I honestly don't see the point of reimplementing some of the things that are already available there, but we can discuss it some other time. For this one, since we already use lodash in a couple of other places in `medusa` I'll leave it as reimplementing this would be unnecessary work, WDYT?

Reply:
but all the utils that we use from lodash can be done easily natively so unless we use complex utils I dont see the need to rely on a third party imo

Reply:
lets do that for now, but imo we should go away from lodash, not the opposite

Reply:
@adrien2p let's create a ticket maybe and make sure we get rid of all references in that case? I'll merge this for now and we can do a cleanup afterwards